src/main/java/wolox/training/controllers/UserApiController.java

```diff
@@ -0,0 +1,70 @@
+package wolox.training.controllers;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import org.springframework.web.bind.annotation.RestController;
+import wolox.training.models.Book;
+import wolox.training.models.User;
+import wolox.training.repositories.BookRepository;
+import wolox.training.repositories.UserRepository;
+
+@RestController
+@RequestMapping("/api/users")
+public class UserApiController extends ApiController {
+
+    @Autowired
+    private UserRepository userRepository;
+    @Autowired
+    private BookRepository bookRepository;
+
+    @PostMapping("/create")
+    @ResponseStatus(HttpStatus.CREATED)
+    public User create(@RequestBody User user) {
```

Comment:
t is considered a good practice in Api Rest not to create specific routes for each method, that is, not to have a path "api/users/create", "api/users/delete", etc.

It is enough to have the different http method to differentiate one url from another. With the same http method you are saying if it is a cration, a search, an update or a deletion, it is not necessary to write it in the url.

I recommend using the following URLs:

POST api/users

DELETE api/users

GET api/ users/{username}

PUT api/books/{username}